cardano-node/src/Cardano/Node/Tracing/Era/HardFork.hs

```diff
@@ -0,0 +1,349 @@
+        , "currentEra" .= ledgerEraName
+        , "txEra"      .= otherEraName
+        ]
+      where
+        EraMismatch {ledgerEraName, otherEraName} = mkEraMismatch mismatch
+
+instance All (LogFormatting `Compose` WrapApplyTxErr) xs => LogFormatting (OneEraApplyTxErr xs) where
+    forMachine dtal =
+        hcollapse
+      . hcmap (Proxy @ (LogFormatting `Compose` WrapApplyTxErr)) (K . forMachine dtal)
+      . getOneEraApplyTxErr
+
+instance LogFormatting (ApplyTxErr blk) => LogFormatting (WrapApplyTxErr blk) where
+    forMachine dtal = forMachine dtal . unwrapApplyTxErr
+
+
+--
+-- instances for HardForkLedgerError
+--
+
+instance All (LogFormatting `Compose` WrapLedgerErr) xs => LogFormatting (HardForkLedgerError xs) where
+    forMachine dtal (HardForkLedgerErrorFromEra err) = forMachine dtal err
+
+    forMachine _dtal (HardForkLedgerErrorWrongEra mismatch) =
+      mkObject
+        [ "kind"       .= String "HardForkLedgerErrorWrongEra"
+        , "currentEra" .= ledgerEraName
+        , "blockEra"   .= otherEraName
+        ]
+      where
+        EraMismatch {ledgerEraName, otherEraName} = mkEraMismatch mismatch
+
+instance All (LogFormatting `Compose` WrapLedgerErr) xs => LogFormatting (OneEraLedgerError xs) where
+    forMachine dtal =
+        hcollapse
+      . hcmap (Proxy @ (LogFormatting `Compose` WrapLedgerErr)) (K . forMachine dtal)
+      . getOneEraLedgerError
+
+instance LogFormatting (LedgerError blk) => LogFormatting (WrapLedgerErr blk) where
+    forMachine dtal = forMachine dtal . unwrapLedgerErr
+
+
+--
+-- instances for HardForkLedgerWarning
+--
+
+instance ( All (LogFormatting `Compose` WrapLedgerWarning) xs
+         , All SingleEraBlock xs
+         ) => LogFormatting (HardForkLedgerWarning xs) where
+    forMachine dtal warning = case warning of
+      HardForkWarningInEra err -> forMachine dtal err
+
+      HardForkWarningTransitionMismatch toEra eraParams epoch ->
+        mkObject
+          [ "kind"            .= String "HardForkWarningTransitionMismatch"
+          , "toEra"           .= condense toEra
+          , "eraParams"       .= forMachine dtal eraParams
+          , "transitionEpoch" .= epoch
+          ]
+
+      HardForkWarningTransitionInFinalEra fromEra epoch ->
+        mkObject
+          [ "kind"            .= String "HardForkWarningTransitionInFinalEra"
+          , "fromEra"         .= condense fromEra
+          , "transitionEpoch" .= epoch
+          ]
+
+      HardForkWarningTransitionUnconfirmed toEra ->
+        mkObject
+          [ "kind"  .= String "HardForkWarningTransitionUnconfirmed"
+          , "toEra" .= condense toEra
+          ]
+
+      HardForkWarningTransitionReconfirmed fromEra toEra prevEpoch newEpoch ->
+        mkObject
+          [ "kind"                .= String "HardForkWarningTransitionReconfirmed"
+          , "fromEra"             .= condense fromEra
+          , "toEra"               .= condense toEra
```

Comment:
I thought we were moving away from `condense` in favour of defining our own JSON instances?

Reply:
I'm not aware of this discussion and just moved the state that I found over.

Reply:
@Jimbo4350, to clarify -- we did not introduce a change in implementation here -- this is mirroring the current implementation in the legacy system.